cmd/registry/controller/parser.go

```diff
@@ -157,39 +158,78 @@ func resourceNameFromParent(
 	return resource, nil
 }
 
-func ValidateResourceEntry(resource *rpc.GeneratedResource) error {
-	var group string
+func ValidateManifest(ctx context.Context, parent string, manifest *rpc.Manifest) []error {
+	totalErrors := make([]error, 0)
+	for _, resource := range manifest.GeneratedResources {
+		if errs := validateGeneratedResourceEntry(parent, resource); len(errs) != 0 {
+			for _, err := range errs {
```

Comment:
Nit: This length check isn't necessary because `range` handles `nil` or empty lists gracefully.

Reply:
Done.